remill/Arch/Runtime/Operators.h

```diff
@@ -1185,6 +1185,14 @@ T _ZeroVec(void) {
       __asm__ __volatile__ ("" ::: "memory"); \
     } while (false)
 
+// A 'compiler' barrier that forced a particular value to be used in a specific
+// spot.
+#define BarrierUsedHere(x) \
```

Comment:
I'm not sure I understand these compiler barriers. `BarrierReorder` prevents the compiler from reordering of instructions before and after it, but does its effect extend to a certain scope around it? If `BarrierReorder` prevents a reordering, shouldn't `BarrierUsedHere` prevent a value from being used (it seems to do the opposite: force it to be used)?

I think what is needed is a one-or-two line comment for each Barrier operator that explains its purpose: when is it needed? Is it to trick the compiler or the direct the CPU?

Nitpick: esthetically -> aesthetically

Reply:
I will document these in my next PR.